### PR TITLE
Splice expression into appropriate stage during v2 metric expansion

### LINKED ISSUE #48722: Metric on a calculated column randomly fails
### Describe the bug

I built a metric that shows the median of a calculated column which then ends up failing when consulted past the first time. To see it again, I need to edit its definition, change the default time dimension to something, and revert it back to see the metric properly computed again.

The model in my case is "Issues" (GitHub) and my calculated column is called "Days open" computed this way `datetimeDiff([Created At], now, "day")`. The metric is then defined as `Median of Days open` (cf. repro steps below).

### To Reproduce

You can check out this repro Loom: https://www.loom.com/share/d7b7926283ea43a7b0eb6cfeafde9714

1. Create a new metric in stats.metabase.com using "Issues" as the source data
2. Create a custom column called "Days open" with the following formula `datetimeDiff([Created At], now, "day")`
3. Define the metric's formula as `Median of Days open`
4. Visualize and save -> Everything works fine
5. Reload Metabase, and navigate to your metric -> There's a failure with the following error message: `Cannot determine the source table or query for Field clause [:field "created_at" {:base-type :type/DateTimeWithLocalTZ}]` 
6. Edit the metric definition
7. Change the default time dimension to anything and click Play -> It works again

### Expected behavior

This should be consistently displayed, without failing.

### Logs

_No response_

### Information about your Metabase installation

Using stats.metabase.com on a Brave browser, but I doubt this has anything to do with the issue :)

### Severity

Prevents from using the metric

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by lbrdnk**
Closes https://github.com/metabase/metabase/issues/48722

### Description

Problem was that expression was added to 0th stage. If metric with source other than table was used, its source was expanded to another stage. This PR adjusts the stage to which expressions are added.

### How to verify

Follow reproduction steps or execute newly added test.

### Checklist

- [X] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: test/metabase/query_processor/middleware/metrics_test.clj [hunk 1]
```diff
@@ -716,3 +716,26 @@
...
+                                                                    [:field "CREATED_AT" {:base-type :type/DateTime}]
+                                                                    [:now] :day]}
+                                          :aggregation [[:median [:expression "somedays" {:base-type :type/Integer}]]]})
+                                       :database_id (mt/id)
+                                       :name "somedays median"
+                                       :type :metric}]
+      (let [mp (lib.metadata.jvm/application-database-metadata-provider (mt/id))
+            query (-> (lib/query mp (lib.metadata/card mp (:id source-model)))
+                      (lib/aggregate (lib.metadata/metric mp (:id metric))))]
+        (is (= 2162
```
Comment (metamben) [inline test/metabase/query_processor/middleware/metrics_test.clj:740]: Isn't this going to change as time goes by?

**Review by metamben — APPROVED** [review]
Should be good once the test gets fixed.

File: test/metabase/query_processor/middleware/metrics_test.clj (hunk 1 above)
Comment (lbrdnk) [reply test/metabase/query_processor/middleware/metrics_test.clj:813]: Thank you, adjusted to fixed datetime.

**Comment by lbrdnk** [comment]
Flaky test has passed locally. Merging.

**Pull request merged**